packages/frontend/src/components/SimpleTable/index.tsx

```diff
@@ -57,28 +57,34 @@ const SimpleTable: FC<SimpleTableProps> = ({
         );
     }
 
-    if (pivotTableData.error || pivotTableData.data) {
-        return pivotTableData.error || !pivotTableData.data ? (
+    if (pivotTableData.error) {
+        return (
             <NonIdealState
                 title="Results not available"
                 description={pivotTableData.error}
                 icon="error"
             />
-        ) : (
+        );
+    } else if (pivotTableData.loading || pivotTableData.data) {
+        return (
             <Box
                 ref={scrollableContainerRef}
                 miw="100%"
                 h="100%"
                 sx={{ overflow: 'auto' }}
             >
-                <PivotTable
-                    containerRef={scrollableContainerRef}
-                    data={pivotTableData.data}
-                    conditionalFormattings={conditionalFormattings}
-                    getFieldLabel={getFieldLabel}
-                    getField={getField}
-                    hideRowNumbers={hideRowNumbers}
-                />
+                {pivotTableData.data ? (
```

Comment:
If `scrollableContainerRef` is null when <PivotTable> mounts, the virtualization won't work as expected.
This was the only way I found to guarantee that `scrollableContainerRef` is not null when we load the pivot table.

Reply:
cool, we could also wrap a check in a `useEffect` to track when it mounts if you want to be more specific, like so

```tsx
useEffect(() => {
    if (scrollableContainerRef.current) {
      setIsScrollableContainerRefMounted(true)
    }
  }, [scrollableContainerRef.current]); 
```

But this works so that's cool!

Reply:
edit: I think you've tried that but it didn't work perhaps 😅

Reply:
mutating the reference won't trigger a render, therefore, won't trigger the useEffect.

Reply:
🤦  of course!